lib/Doctrine/DBAL/Statement.php

```diff
@@ -135,6 +136,10 @@ public function bindParam($param, &$variable, $type = ParameterType::STRING, $le
         $this->params[$param] = $variable;
         $this->types[$param]  = $type;
 
+        if ($this->stmt instanceof PDOStatement) {
+            $length = $length ?? 0;
```

Comment:
This looks like a BC break regardless of the version.

Reply:
It isn't. The `$length` parameter has to be an `int`. If you pass `null`, it is silently cast to zero on php 7. On php 8 you'll get a type error. If you prefer, I can additionally check if we're on php 8, but I don't think that this is necessary.

Reply:
There can be a class that extends the PDO statement which this code will call and it may expect NULL.

Reply:
That class would be violating the contract of `PDOStatement` if `null` led to a different (non-exceptional) behavior than zero. But yes, it's possible. Do you have a suggestion how to move forward here?

Reply:
> The `$length` parameter has to be an `int`.

You're right. I was under the impression that it became non-nullable in PHP 8. No additional changes needed.